DeBugIt/Game.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+//#include "GameStateManager.h"
+
+enum class GameStateId
+{
+	TitleScreen,
+	Playing,
+	PauseMenu,
+	WinMenu,
+	LoseMenu,
+	TowerMenu,
```

Comment:
TowerMenu should be a one of the part of Playing state. I believe you don't want to freeze your playing state while in TowerMenuState. There's nothing wrong in creating other states in one of the states.